assets/prototype/domain/shared/predicates/priceTypes/selectionPredicates/index.test.ts

```diff
@@ -0,0 +1,127 @@
+/**
+ * Internal dependencies
+ */
+import { PriceBasetype } from '../../../../../domain/eventEditor/data/types';
+import {
+	isBasePrice,
+	isNotBasePrice,
+	isDiscount,
+	isNotDiscount,
+	isPercent,
+	isNotPercent,
+	isTax,
+	isNotTax,
+	isFlatFeeSurcharge,
+	getDefaultPriceModifierType,
+} from './index';
+import { nodes as priceTypes } from '../../../../../domain/eventEditor/data/queries/priceTypes/test/data';
+
+describe('isBasePrice & isNotBasePrice', () => {
+	it('should return true if priceType is base price', () => {
+		priceTypes.forEach((priceType) => {
+			if (priceType.isBasePrice === true) {
+				expect(isBasePrice(priceType)).toBe(true);
+			}
+		});
+	});
+
+	it('should return true if priceType is NOT base price', () => {
+		priceTypes.forEach((priceType) => {
+			if (priceType.isBasePrice === false) {
+				expect(isNotBasePrice(priceType)).toBe(true);
+			}
+		});
+	});
+});
+
+describe('isDiscount & isNotDiscount', () => {
+	it('should return true if priceType is a discount', () => {
+		priceTypes.forEach((priceType) => {
+			if (priceType.isDiscount === true) {
+				expect(isDiscount(priceType)).toBe(true);
+			}
+		});
+	});
+
+	it('should return true if priceType is NOT a discount', () => {
+		priceTypes.forEach((priceType) => {
+			if (priceType.isDiscount === false) {
+				expect(isNotDiscount(priceType)).toBe(true);
+			}
+		});
+	});
+});
+
+describe('isPercent & isNotPercent', () => {
+	it('should return true if priceType is percent', () => {
+		priceTypes.forEach((priceType) => {
+			if (priceType.isPercent === true) {
+				expect(isPercent(priceType)).toBe(true);
+			}
+		});
+	});
+
+	it('should return true if priceType is NOT percent', () => {
+		priceTypes.forEach((priceType) => {
+			if (priceType.isPercent === false) {
+				expect(isNotPercent(priceType)).toBe(true);
+			}
+		});
+	});
+});
+
+describe('isTax & isNotTax', () => {
+	it('should return true if priceType is tax', () => {
+		priceTypes.forEach((priceType) => {
+			if (priceType.isTax === true) {
+				expect(isTax(priceType)).toBe(true);
+			}
+		});
+	});
+
+	it('should return true if priceType is NOT tax', () => {
+		priceTypes.forEach((priceType) => {
+			if (priceType.isTax === false) {
+				expect(isNotTax(priceType)).toBe(true);
+			}
+		});
+	});
+});
+
+describe('isFlatFeeSurcharge', () => {
+	it('should return true if price type is flat fee surcharge', () => {
+		priceTypes.forEach((priceType) => {
+			if (priceType.isBasePrice === false && priceType.isDiscount === false && priceType.isPercent === false) {
+				expect(isFlatFeeSurcharge(priceType)).toBe(true);
+			}
```

Comment:
this test has at least three potential points of failure since it is testing against three properties as well as the logic that combines those into one comparison, so if any of the tests used internally by `isFlatFeeSurcharge()` fails, then it could be giving false positives, but this unit test doesn't check for that.

this one could capture the result of the conditional into a var and then compare against that

ie:

```js
	const dollarSurcharge = priceType.isBasePrice === false && priceType.isDiscount === false && priceType.isPercent === false;
	expect(isFlatFeeSurcharge(priceType)).toBe(dollarSurcharge);
```

[EDIT]

oops... just saw below that you are also testing all of the property combinations separately, so my above comment is not entirely accurate at all since you are testing other possibilities.

that said, I still don't think it's wise to apply tests conditionally like this since you are potentially missing out on an opportunity to catch some weird edge case failure.